lib/board.rb

```diff
@@ -0,0 +1,47 @@
+require 'pry'
+class Board
+  attr_accessor :cells
+  
+  def reset!
+    self.cells = Array.new(9, " ")
+  end
+  
+  def initialize
+    self.cells = Array.new(9, " ")
+  end
+  
+  def display
+   puts " #{self.cells[0]} | #{self.cells[1]} | #{self.cells[2]} "
+    puts "-----------"
+    puts " #{self.cells[3]} | #{self.cells[4]} | #{self.cells[5]} "
+    puts "-----------"
+    puts " #{self.cells[6]} | #{self.cells[7]} | #{self.cells[8]} "
+  end
+  
+  def position(num)
+    input = num.to_i - 1 
+    self.cells[input]
+  end
+  
+  def full?
+    self.cells.all? {|c| c != " " && c != ""}
+  end 
+  
+  def turn_count
+    count = 0 
```

Comment:
Use the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count { |cell| cell == "X" || cell == "O" }
end
```